broker/query_plan_non_agg.go

```diff
@@ -140,6 +145,10 @@ func (nqp *NonAggQueryPlan) Execute(ctx context.Context) (err error) {
 	}
 
 	for i := 0; i < len(nqp.nodes); i++ {
+		if nqp.getRowsWanted() == 0 {
```

Comment:
better use notification or channel to find ready nqp to process return data immediately, instead of looping through nqps.

Reply:
this is not really looping through nodes.. it's like a while loop, but we know before hand the max number of time we go into the loop will be number of child nodes.
the blocking channel read is at line 153